src/Symfony/Component/Security/Http/Tests/Firewall/SessionExpirationListenerTest.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Tests\Firewall;
+
+use Symfony\Component\Security\Http\Firewall\SessionExpirationListener;
+
+/**
+ * @author Antonio J. García Lagar <aj@garcialagar.es>
+ */
+class SessionExpirationListenerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testHandleWhenNoSession()
+    {
+        $request = $this->getMock('Symfony\Component\HttpFoundation\Request');
+        $request
+            ->expects($this->once())
+            ->method('getSession')
+            ->will($this->returnValue(false));
+
+        $event = $this->getMock('Symfony\Component\HttpKernel\Event\GetResponseEvent', array(), array(), '', false);
```

Comment:
use ``getMockBuilder``. It will be much more readable than passing all the arguments to ``getMock`` (I don't know what the order of the getMock arguments is, and I doubt lots of people are knowing it without checking the doc given the complexity of the signature, which is precisely why there is a mock builder)